src/Data/Fin/Relation/Unary/Top.agda

```diff
@@ -0,0 +1,135 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The 'top' view of Fin
+--
+-- This is an example of a view of (elements of) a datatype,
+-- here i : Fin (suc n), which exhibits every such i as
+-- * either, i = fromℕ n
+-- * or, i = inject₁ j for a unique j : Fin n
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Fin.Relation.Unary.Top where
+
+open import Data.Nat.Base using (ℕ; zero; suc; _<_)
+open import Data.Fin.Base using (Fin; zero; suc; toℕ; fromℕ; inject₁)
+open import Data.Fin.Properties using (toℕ-fromℕ; toℕ-inject₁; inject₁ℕ<)
+open import Relation.Binary.PropositionalEquality
+
+private
+  variable
+    n : ℕ
+
+------------------------------------------------------------------------
+-- The View, considered as a unary relation on Fin (suc n)
+
+-- NB `Data.Fin.Properties.fromℕ≢inject₁` establishes that the following
+-- inductively defined family on `Fin (suc n)` has constructors which
+-- target *disjoint* instances of the family; and hence the interpretations
+-- of the View constructors will also be disjoint
+
+data View : (i : Fin (suc n)) → Set where
+
+  top :                View (fromℕ n)
+  inj : (j : Fin n) → View (inject₁ j)
```

Comment:
Should this come with a further `View j`? We're already traversing the whole `Fin`
so may as well hand users the ability to repeatedly use the view without redoing
the work?

Reply:
Hmmm... making the proposed change mangles the index arithmetic... with one outstanding proof in `Instances` failing to through... without my thinking a bit harder.

Reply:
Hmmm... this may in fact be a simplification... more to folloe

Reply:
Yes, I think I agree with you, so will push commits which:
* make the view recursive
* add a 'non-recursive' smart constructor/pattern synonym which projects out the `j : Fin n` component
* simplify with types + examples

And which... now makes this a view of `Fin n` (!)